src/app/features/increase-fee-drawer/hooks/use-btc-increase-fee.ts

```diff
@@ -0,0 +1,167 @@
+import { queryClient } from '@app/common/persistence';
+import {
+  getBitcoinTxValue,
+  getRecipientAddressFromOutput,
+  getSizeInfo,
+} from '@app/common/transactions/bitcoin/utils';
+import { MAX_FEE_RATE_MULTIPLIER } from '@app/components/bitcoin-custom-fee/hooks/use-bitcoin-custom-fee';
+import { useBitcoinFeesList } from '@app/components/bitcoin-fees-list/use-bitcoin-fees-list';
+import { useCurrentNativeSegwitUtxos } from '@app/query/bitcoin/address/utxos-by-address.hooks';
+import { useBitcoinBroadcastTransaction } from '@app/query/bitcoin/transaction/use-bitcoin-broadcast-transaction';
+import { useBitcoinScureLibNetworkConfig } from '@app/store/accounts/blockchain/bitcoin/bitcoin-keychain';
+import { useCurrentAccountNativeSegwitIndexZeroSigner } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
+
+export function useBtcIncreaseFee(btcTx: BitcoinTx) {
+  const navigate = useNavigate();
+  const networkMode = useBitcoinScureLibNetworkConfig();
+  const analytics = useAnalytics();
+
+  const {
+    address: currentBitcoinAddress,
+    sign,
+    publicKeychain: currentAddressIndexKeychain,
+  } = useCurrentAccountNativeSegwitIndexZeroSigner();
+  const { data: utxos = [], refetch } = useCurrentNativeSegwitUtxos();
+  const { broadcastTx, isBroadcasting } = useBitcoinBroadcastTransaction();
+  const recipient = getRecipientAddressFromOutput(btcTx.vout, currentBitcoinAddress) || '';
+  const sizeInfo = getSizeInfo({
+    inputLength: btcTx.vin.length,
+    recipient,
+    outputLength: btcTx.vout.length,
+  });
+
+  const { btcAvailableAssetBalance } = useBtcAssetBalance(currentBitcoinAddress);
+  const sendingAmount = getBitcoinTxValue(currentBitcoinAddress, btcTx);
+  const { feesList } = useBitcoinFeesList({
+    amount: Number(sendingAmount),
+    isSendingMax: false,
+    recipient,
+    utxos,
+  });
+
+  function generateTx(payload: { feeRate: string; tx: BitcoinTx }) {
+    const newTx = new btc.Transaction();
+    const { vin, vout, fee: prevFee } = payload.tx;
+    const p2wpkh = btc.p2wpkh(currentAddressIndexKeychain.publicKey!, networkMode);
+
+    vin.forEach(input => {
+      newTx.addInput({
+        txid: input.txid,
+        index: input.vout,
+        sequence: input.sequence + 1,
+        witnessUtxo: {
+          // script = 0014 + pubKeyHash
+          script: p2wpkh.script,
+          amount: BigInt(input.prevout.value),
+        },
+      });
+    });
+
+    const newFee = Math.ceil(sizeInfo.txVBytes * Number(payload.feeRate));
+    const feeDiff = newFee - prevFee;
+
+    vout.forEach(output => {
+      if (output.scriptpubkey_address === currentBitcoinAddress) {
+        const outputDiff = output.value - feeDiff;
+
+        if (outputDiff < 0) {
+          void analytics.track('increase_fee_error', {
+            outputDiff,
+          });
+          throw new Error('Previous tx inputs cannot cover new fee');
+        }
```

Comment:
@kyranjamie smth like this?

Reply:
If it's 0 or 1 can the fee be covered? 

Suggest `bitcoin_rbf_fee_increase_error` as name

Reply:
I think yes 🤔 0 means nothing is sent back to you and all goes to miner

Reply:
checked, and it worked ok